crates/dojo-world/src/manifest.rs

```diff
@@ -121,118 +104,197 @@ pub struct Contract {
-    pub fn load_from_path(path: impl AsRef<Path>) -> Result<Self, std::io::Error> {
-        let file = fs::File::open(path)?;
-        Ok(Self::try_from(file)?)
+    pub fn class_hash(&self) -> FieldElement {
+        match &self.kind {
+            ManifestKind::Class(class) => class.class_hash,
+            ManifestKind::Contract(contract) => contract.class_hash,
+            ManifestKind::Model(model) => model.class_hash,
+        }
+    }
+
+    pub fn abi(&self) -> Option<&String> {
+        match &self.kind {
+            ManifestKind::Class(class) => class.abi.as_ref(),
+            ManifestKind::Contract(contract) => contract.abi.as_ref(),
+            ManifestKind::Model(model) => model.abi.as_ref(),
+        }
     }
 
-    /// Writes the manifest into a file at the given path. Will return error if the file doesn't
-    /// exist.
-    pub fn write_to_path(self, path: impl AsRef<Path>) -> Result<(), std::io::Error> {
-        let fd = fs::File::options().write(true).open(path)?;
-        Ok(serde_json::to_writer_pretty(fd, &self)?)
+    pub fn set_class_hash(&mut self, class_hash: FieldElement) {
+        match &mut self.kind {
+            ManifestKind::Class(class) => class.class_hash = class_hash,
+            ManifestKind::Contract(contract) => contract.class_hash = class_hash,
+            ManifestKind::Model(model) => model.class_hash = class_hash,
+        }
     }
 
-    /// Construct a manifest of a remote World.
-    ///
-    /// # Arguments
-    /// * `provider` - A Starknet RPC provider.
-    /// * `world_address` - The address of the remote World contract.
-    pub async fn load_from_remote<P>(
+    pub fn set_abi(&mut self, abi: Option<String>) {
+        match &mut self.kind {
+            ManifestKind::Class(class) => class.abi = abi,
+            ManifestKind::Contract(contract) => contract.abi = abi,
+            ManifestKind::Model(model) => model.abi = abi,
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum ManifestKind {
+    Class(Class),
+    Contract(Contract),
+    Model(Model),
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct World {
+    pub world: Manifest,
+    pub executor: Manifest,
+    pub base: Manifest,
+    pub contracts: Vec<Manifest>,
+    pub models: Vec<Manifest>,
+}
+
+impl World {
+    // here expected path is path to the folder that contains manifest file
+    pub fn load_deployed_from_path(path: Utf8PathBuf) -> Result<Self, std::io::Error> {
+        let base_dir = Utf8PathBuf::new().join(path).join("manifest").join("deployed");
+        Self::load_from_path(base_dir)
+    }
+
+    // here expected path is path to the folder that contains manifest file
+    pub fn load_base_from_path(path: Utf8PathBuf) -> Result<Self, std::io::Error> {
+        let base_dir = Utf8PathBuf::new().join(path).join("manifest").join("base");
+        Self::load_from_path(base_dir)
```

Comment:
Base here seems to be loaded from a `base` folder. But in `load_from_path` it looks like base is at the same level as `world` and `executor` without actually using the `base` subfolder.

Reply:
`base` here is overloaded between base manifests (i.e. the folder where we output manifests during `sozo build`) and `base` contract used by dojo. naming is hard :sweat_smile: